### PR TITLE
Version 0.4.0 Rewrite

### PR DESCRIPTION
**Pull request by reteps**

### CODE REVIEW

File: secedgar/filings/_base.py [hunk 1]
```diff
@@ -5,39 +5,18 @@
...
-        """Extract meta data from filings in directory."""
-        for root, _, files in os.walk(directory):
-            for file in files:
-                if file.endswith('.txt'):
-                    MetaParser().process(os.path.join(root, file),
-                                         out_dir=out_dir,
-                                         create_subdir=create_subdir,
-                                         rm_infile=rm_infile)
+    def __init__(self, **kwargs):
+        self._client = kwargs.get('client', NetworkClient())
```
Comment (jackmoody11) [inline secedgar/filings/_base.py:14]:
```suggestion
    def __init__(self, client=None, **kwargs):
        self._client = client or NetworkClient(**kwargs)
```

File: secedgar/filings/_index.py [hunk 2]
```diff
@@ -2,18 +2,19 @@
 import os
 import re
 import shutil
+import tempfile
```
Comment (jackmoody11) [inline secedgar/filings/_index.py:5]: Didn't know about this builtin. Very cool!

File: secedgar/utils/cik_map.py [hunk 3]
```diff
@@ -1,25 +0,0 @@
-import json
```
Comment (jackmoody11) [inline secedgar/utils/cik_map.py:1]: Why did you delete this file?

Comment (reteps) [reply secedgar/utils/cik_map.py:1]: I moved the CIK Map into the `CIKLookup`.
